base/compiler/optimize.jl

```diff
@@ -175,7 +190,7 @@ function stmt_affects_purity(@nospecialize(stmt), ir)
 end
 
 # Convert IRCode back to CodeInfo and compute inlining cost and sideeffects
-function finish(opt::OptimizationState, params::OptimizationParams, ir, @nospecialize(result))
+function finish(interp::AbstractInterpreter, opt::OptimizationState, params::OptimizationParams, ir, @nospecialize(result))
```

Comment:
Looks like this argument is unused? Just added for consistency?

Reply:
In an earlier version of this code I needed it, but this turned out to be better. However, this function has a lot of logic, so it can be useful for debugging to hook it for a particular AbstractInterpreter to play with that logic, so I kept it.